zenodo/modules/deposit/api.py

```diff
@@ -213,3 +218,39 @@ def _publish_edited(self):
         if not record['communities']:
             del record['communities']
         return record
+
+    @staticmethod
+    def _build_agent_info():
+        """Build the SIP agent info."""
+        email = ''
+        ip_address = ''
+        if current_user.is_authenticated:
+            if request.remote_addr:
+                ip_address = request.remote_addr
+            if current_user.email:
+                email = current_user.email
+        else:
+            return None
+        return dict(email=email, ip_address=ip_address)
+
+    def _create_sips(self, pid, record):
+        """Create the SIP information for the record."""
+        user_id = None if current_user.is_anonymous else current_user.id
+        agent = self._build_agent_info()
+        sip = SIP.create('json', json.dumps(record.dumps()), user_id=user_id,
+                         agent=agent)
+        recsip = RecordSIP(sip_id=sip.id, pid_id=pid.id)
```

Comment:
Perhaps we can move some of the code to ``RecordSIP`` or make a ``ZenodoRecordSIP`` which knowns how to make a SIP out of a Zenodo record. We can discuss IRL